mjx/mujoco/mjx/_src/solver.py

```diff
@@ -590,7 +590,13 @@ def body(ctx: Context) -> Context:
   if m.opt.iterations == 1:
     ctx = body(ctx)
   else:
-    ctx = jax.lax.while_loop(cond, body, ctx)
+    # if tolerance is 0, scan is differentiable. otherwise, use while_loop.
+    ctx = jax.lax.cond(
+        m.opt.tolerance == 0,
```

Comment:
@varshneydevansh it's probably better to check this statically.  It should produce better run-time performance.  So something like this:

```py
if m.opt.iterations == 1:
  ctx = body(ctx)
elif m.opt.tolerance == 0.0:
  ctx = _while_loop_scan(cond, body, ctx, m.opt.iterations)
else:
  ctx = jax.lax.while_loop(cond, body, ctx)
```